Feat/database status (#98)
digidem/mapfilter-db: README.md

Retrieves the current device's unique ID (string).

### `status = mapeo.getDatabaseStatus()`

> Retrieves the status of the database, returns an array of hypercore feed

Reviewer: ```suggestion
Retrieves the status of the database, returns an array of Device IDs and their download status.
```

Reviewer: I was thinking (as per discussions /w @gmaclennan) we could try and use higher-level terminology in the mapeo-* modules, and do our best to keep hypercore-y terminology under the API covers. Does this sound ok with you?

> download status.

Reviewer: ```suggestion
```

```

> [{

Reviewer: @okdistribute What do you think about this being a map on the outside, so we have room later to extend for indexing status?

```js
{
  devices: [ ... ]  // same as you have here
  indexes: { ... }  // content to-be-determined
}
```

Reviewer: Alternately, we could have separate `getDevicesStatus()` and `getIndexesStatus()` APIs. What do you think?

cc @gmaclennan

Author: We could do both :)

Reviewer 2: I prefer separate functions (`getDevicesStatus()` and `getIndexesStatus()`) so that the client can choose what they need. I'm not a fan of both, because it adds API surface area without adding functionality, just convenience (just calling one method instead of two). If we eventually make the API promise based it becomes easy:

```js
const [devices, indexes] = await Promise.all(getDevicesStatus(), getIndexesStatus())
```

Reviewer: I think either will work well enough. I feel good about @okdistribute just making the call based on our arguments versus us bikeshedding it. :+1:

Author: thanks for the input all,

because this is a small api, I'm going to go with the simple approach I had in the beginning, but rename to `getFeedStatus` to be more clear we're talking about the feeds. When we add media and indexing support, we can add those functions as well

>   id: DeviceID,

Reviewer: ```suggestion
  id: DeviceID (String[64]),
```

Retrieves the current device's unique ID (string).

### `mapeo.getDatabaseStatus(cb)`

Retrieves the status of the database, returns an array of Device IDs and their download status.

```
{ 

>   devices: [{

Reviewer 2: At some point we might want to add support for media items too in this API, so I think we should prefix the sofar and total with something e.g. `dbTotal`. We could also nest this e.g. `db.total` but nested structures are always a pain to deal with in JS just because of the risk of `undefined` props throwing an error.

    id: DeviceID (String[64]),

>     sofar: Number,

Reviewer 2: I think that `dbDownloaded` would be clearer to a developer than `sofar`. If we do support sparse and allow intentional removal of data from a device (either for disk space or security reasons) the concept of "I have downloaded X records out of Y total" makes more sense to me.

Author: `sofar` is currently what is passed in through progress so I wanted to keep it uniform.

What about `database` or `feeds` instead of `devices` ?

Author: going with `feeds`
